engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -766,156 +802,72 @@ public function fetchFromSql($sql, \ElggBatch $batch = null) {
 	 * @access private
 	 */
 	public function getEntityTypeSubtypeWhereSql($table, $types, $subtypes, $pairs) {
-		// subtype depends upon type.
-		if ($subtypes && !$types) {
-			$this->logger->warn("Cannot set subtypes without type.");
-			return false;
-		}
-
 		// short circuit if nothing is requested
 		if (!$types && !$subtypes && !$pairs) {
 			return '';
 		}
 
-		// pairs override
-		$wheres = [];
-		if (!is_array($pairs)) {
-			if (!is_array($types)) {
-				$types = [$types];
-			}
+		// these are the only valid types for entities in elgg
+		$valid_types = \Elgg\Config::getEntityTypes();
```

Comment:
If you no longer wish to validate `$types` than please remove this part, as it says you're going to validate the `$types`